src/store/modules/product/actions.ts

```diff
@@ -40,6 +42,37 @@ const actions: ActionTree<ProductState, RootState> = {
     return resp;
   },
 
+  async findProduct ({ commit, state }, payload) {
+    // Show loader only when new query and not the infinite scroll
+    if (payload.viewIndex === 0) emitter.emit("presentLoader");
+    let resp;
+    try {
+      resp = await ProductService.fetchProducts({
```

Comment:
Implement a separate service for findProducts

Reply:
Implemented sir